src/modules/file/file.usecase.ts

```diff
@@ -321,6 +322,66 @@ export class FileUseCases {
     };
   }
 
+  async moveFile(
+    user: User,
+    fileUuid: File['fileId'],
+    destinationUuid: File['folderUuid'],
+  ): Promise<File> {
+    const file = await this.fileRepository.findByUuid(fileUuid, user.id);
+    if (!file) {
+      throw new NotFoundException(`File ${fileUuid} not found`);
+    }
+
+    const destinationFolder = await this.folderUsecases.getFolderByUuidAndUser(
+      destinationUuid,
+      user,
+    );
+    if (!destinationFolder) {
+      throw new NotFoundException(`Folder ${destinationUuid} not found`);
+    }
+
```

Comment:
Ensure the origin `folder.removed = false`. We cannot allow to move files from folders that are removed permanently